world/boards/views.py

```diff
@@ -1,18 +1,66 @@
+import re
 from django.shortcuts import get_object_or_404, render
-from world.boards.models import Board
+from django.http import HttpResponseNotFound
+from world.boards.models import Board, Post
 
 # List page for all boards.
 def boards(request):
     all_boards = Board.objects.all()
     context = { "all_boards": all_boards }
     return render(request, "boards/list.html", context)
 
-def detail(request, post_id):
-    board = get_object_or_404(Board, pk=post_id)
+# Shows a board detail page which doubles as a post detail page. Somewhat like a playlist
+# view in a media app, we show one focused post in detail while also showing the list of
+# all posts in the board in chronological order.
+def detail(request, board_id, post_id=None):
+    board = get_object_or_404(Board, pk=board_id)
```

Comment:
Hey dummy...you forgot to actually mark the post as read by the current user. Maybe you should do that?

Reply:
Ah yeah, good call boss. Will do that now.

Reply:
Done!